Multiple hosted Blazor apps get 404 error on blazor.webassembly.js

### Describe the bug

I'm trying to create a Razor pages app with multiple Blazor "sub apps" at urls like /app1 and /app2. Everything works fine locally when debugging in Visual Studio but when I publish and try to run locally using the dotnet command (or when I try to host behind an Nginx reverse proxy) I get a 404 error for _framework/blazor.webassembly.js.

Thanks for any help you can offer.


### To Reproduce

Below are my steps and the simplest possible repro is at the following public repo:

https://github.com/fuzl-llc/MultipleHostedBlazorApps

In Visual Studio 2019, created project MultipleHostedBlazorApps with default .NET 5.0 "Blazor WebAssembly App" template. Deselected "Configure for HTTPS" and checked "ASP.NET Core hosted."
Checked in.


Added second client project called "Client2" with HTTPS and ASP.NET Core Hosted unchecked.
The project, MultipleHostedBlazorApps.Client2, was put in a "Client2" folder next to the existing "Client" folder.
Checked in.


Deleted all IIS references from all three project's launchSettings.json. Left server app to run on port 5000, updated client and client 2 to run on ports 5001 and 5002, respectively. In retrospect I don't think this matters as all projects load under the host app's port 5000...

Updated Blazor apps index.html files to have &lt;base href="/app1/" /> and &lt;base href="/app2/" /> for client and client2 (and matching page titles).

Started getting: Conflicting assets with the same path '/wwwroot/css/app.css' for content root paths error (but below actions fix it).

Changed each Blazor app's project file to include:
&lt;StaticWebAssetBasePath>app1&lt;/StaticWebAssetBasePath> (and &lt;StaticWebAssetBasePath>app2&lt;/StaticWebAssetBasePath>)
in the top property group under &lt;TargetFramework>net5.0&lt;/TargetFramework>

In main host app's startup file, mapped both Blazor apps using a new extension method called "MapBlazorApp" which uses "MapWhen" to set up each Blazor app to be loaded at "sub app urls" of the main host app, at /app1 and /app2 of the main host app. This does things like UseBlazorFrameworkFiles passing in the new paths - see code for details.

Added an "index" Razor page to the host app with links to the two sub apps. Added links in both Blazor "sub apps" to each other and the host app main page, updated both app's index page title to show which app they are.

We can now run the app, see both apps working, and click between them.

Checked in.


This seems to be "success" as everything works beautifully when running locally in Visual Studio.  The problems come now when I try to deploy... So, we right click and publish the host app.  As part of this we change the default folder profile path from bin\Release\net5.0\publish\ to "C:\Temp\Multi." In that path we then see, under C:\Temp\Multi\wwwroot, folders like app1, app1_framework, app1app1, app1css which at first seems to make sense e.g. the prefix to distinguish framework files but the app1app1 folder seems a bit strange.

I can run the following successfully:

C:\Temp\Multi>dotnet MultipleHostedBlazorApps.Server.dll --urls http://localhost:5000

Everything appears great at first - I see the host app homepage with links to the two sub apps and when I click them I see the standard Blazor "Loading..." message but when I look in Chrome dev tools > Network, I see that http://localhost:5000/app1/_framework/blazor.webassembly.js gets a 404 "Not Found" error and the app never loads. The same thing happens for http://localhost:5000/app2/_framework/blazor.webassembly.js.  All other static files seem fine, such as http://localhost:5000/app1/css/bootstrap/bootstrap.min.css and other .css files.  If I search for blazor.webassembly.js I find it exists under app1_framework and app2_framework (as well as other versions with .br and .gz extensions appended).

Ultimately I'm trying to host this on Ubuntu via Nginx reverse proxy where I'd set up a config something like this:


server
{
    listen 443 ssl;
    server_name test.com;
    ssl_certificate /etc/letsencrypt/live/test.com/fullchain.pem;
    ssl_certificate_key /etc/letsencrypt/live/test.com/privkey.pem;
    include /etc/letsencrypt/options-ssl-nginx.conf;
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem;

    location /
    {
        proxy_pass         http://localhost:5000;
        proxy_http_version 1.1;
        proxy_set_header   Upgrade $http_upgrade;
        proxy_set_header   Connection keep-alive;
        proxy_set_header   Host $host;
        proxy_cache_bypass $http_upgrade;
        proxy_set_header   X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header   X-Forwarded-Proto $scheme;
    }
    # works except _framework/blazor.webassembly.js gets returned as the HTML of the root of the site:
    location /app1 {
        root /var/www/test.com/wwwroot;
        try_files $uri $uri/ /app1/index.html;
    }

    location /app2 {
        root /var/www/test.com/wwwroot;
        try_files $uri $uri/ /app2/index.html;
    }
}
server
{
    listen 80;
    server_name test.com;
    return 301 https://$host$request_uri;
}


It seemed to work similarly where all static (html, css) files work but the blazor.webassembly.js file gives a 404.

How get I get this blazor.webassembly.js file to load properly using (1) the dotnet command locally and (2) hosting behind an Nginx reverse proxy?




### Exceptions (if any)
Just a 404 error that blazor.webassembly.js cannot be found.




### Further technical details

- ASP.NET Core version:

5.0.101

- Include the output of `dotnet --info`

.NET SDK (reflecting any global.json):
 Version:   5.0.101
 Commit:    d05174dc5a

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.18363
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.101\

Host (useful for support):
  Version: 5.0.1
  Commit:  b02e13abab

.NET SDKs installed:
  3.1.301 [C:\Program Files\dotnet\sdk]
  3.1.400 [C:\Program Files\dotnet\sdk]
  5.0.101 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-rc.1.20451.14 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version
Visual Studio 2019 v16.8.3

Steps:

1. Get the latest code from my sample repo at https://github.com/fuzl-llc/MultipleHostedBlazorApps
2. If you already had the code, clean solution and/or delete all bin/obj folders.
3. Open cmd window
4. cd [your-path]\MultipleHostedBlazorApps\Server
5. dotnet publish -c Debug
6. Everything seems to publish fine, including putting _framework files in correct places.
7. Open new command window
8. cd [your-path]\MultipleHostedBlazorApps\Server\bin\Debug\net5.0\publish
9. dotnet MultipleHostedBlazorApps.Server.dll
10. Point browser to http://localhost:5000 and click on "App1" link (or just point browser to http://localhost:5000/app1)
11. Look in Chrome dev tools and notice in Console tab:

blazor.webassembly.js:1 Error parsing 'integrity' attribute ('undefined'). The hash algorithm must be one of 'sha256', 'sha384', or 'sha512', followed by a '-' character.
(anonymous) @ blazor.webassembly.js:1
blazor.webassembly.js:1 Uncaught (in promise) Error: Content hash is required
    at e.<anonymous> (blazor.webassembly.js:1)
    at blazor.webassembly.js:1
    at Object.next (blazor.webassembly.js:1)
    at blazor.webassembly.js:1
    at new Promise (<anonymous>)
    at r (blazor.webassembly.js:1)
    at e.loadResourceWithCaching (blazor.webassembly.js:1)
    at e.loadResource (blazor.webassembly.js:1)
    at blazor.webassembly.js:1
    at l (blazor.webassembly.js:1)
blazor.webassembly.js:1 Uncaught (in promise) Error: Content hash is required
    at e.<anonymous> (blazor.webassembly.js:1)
    at blazor.webassembly.js:1
    at Object.next (blazor.webassembly.js:1)
    at blazor.webassembly.js:1
    at new Promise (<anonymous>)
    at r (blazor.webassembly.js:1)
    at e.loadResourceWithCaching (blazor.webassembly.js:1)
    at e.loadResource (blazor.webassembly.js:1)
    at blazor.webassembly.js:1
    at l (blazor.webassembly.js:1)
undefined:1 

12. Notice in Chrome dev tools Network tab that while blazor.webassembly.js and blazor.boot.json load properly there is a 404 for a request to http://localhost:5000/app1/_framework/undefined
13. The Blazor app just hangs and never fully loads.

--- a/src/Components/WebAssembly/Sdk/integrationtests/WasmPublishIntegrationTest.cs
+++ b/src/Components/WebAssembly/Sdk/integrationtests/WasmPublishIntegrationTest.cs
@@ -970,6 +970,58 @@ $@"<PropertyGroup>
             Assert.FileDoesNotExist(result, publishOutputDirectory, "wwwroot", "_framework", "icudt_no_CJK.dat");
         }
 
+        [Fact]
+        public async Task Publish_HostingMultipleBlazorWebApps_Works()
+        {
+            // Regression test for https://github.com/dotnet/aspnetcore/issues/29264
+            // Arrange
+            using var project = ProjectDirectory.Create("BlazorMultipleApps.Server", "BlazorMultipleApps.FirstClient", "BlazorMultipleApps.SecondClient", "BlazorMultipleApps.Shared");
+            var result = await MSBuildProcessManager.DotnetMSBuild(project, "Publish");
+
+            var publishOutputDirectory = project.PublishOutputDirectory;
+
+            Assert.FileExists(result, Path.Combine(publishOutputDirectory, "BlazorMultipleApps.Server.dll"));
+            Assert.FileExists(result, Path.Combine(publishOutputDirectory, "BlazorMultipleApps.FirstClient.dll"));
+            Assert.FileExists(result, Path.Combine(publishOutputDirectory, "BlazorMultipleApps.SecondClient.dll"));
+
+            var firstAppPublishDirectory = Path.Combine(publishOutputDirectory, "wwwroot", "FirstApp");
+
+            var firstCss = Assert.FileExists(result, Path.Combine(firstAppPublishDirectory, "css", "app.css"));
+            Assert.FileContains(result, firstCss, "/* First app.css */");
+
+            var firstBootJsonPath = Path.Combine(firstAppPublishDirectory, "_framework", "blazor.boot.json");
+            var firstBootJson = ReadBootJsonData(result, firstBootJsonPath);
+
+            // Do a sanity check that the boot json has files.
+            Assert.Contains("System.Text.Json.dll", firstBootJson.resources.assembly.Keys);
+
+            VerifyBootManifestHashes(result, firstAppPublishDirectory);
+
+            // Verify compression works
+            Assert.FileExists(result, firstAppPublishDirectory, "_framework", "dotnet.wasm.br");
+            Assert.FileExists(result, firstAppPublishDirectory, "_framework", "BlazorMultipleApps.FirstClient.dll.br");
+            Assert.FileExists(result, firstAppPublishDirectory, "_framework", "System.Text.Json.dll.br");
+
+            var secondAppPublishDirectory = Path.Combine(publishOutputDirectory, "wwwroot", "SecondApp");
+
+            var secondCss = Assert.FileExists(result, Path.Combine(secondAppPublishDirectory, "css", "app.css"));
+            Assert.FileContains(result, secondCss, "/* Second app.css */");
+
+            var secondBootJsonPath = Path.Combine(secondAppPublishDirectory, "_framework", "blazor.boot.json");
+            var secondBootJson = ReadBootJsonData(result, secondBootJsonPath);
+
+            // Do a sanity check that the boot json has files.
+            Assert.Contains("System.Private.CoreLib.dll", secondBootJson.resources.assembly.Keys);
+
+            VerifyBootManifestHashes(result, secondAppPublishDirectory);
+
+            // Verify compression works
+            Assert.FileExists(result, secondAppPublishDirectory, "_framework", "dotnet.wasm.br");
+            Assert.FileExists(result, secondAppPublishDirectory, "_framework", "BlazorMultipleApps.SecondClient.dll.br");
+            Assert.FileExists(result, secondAppPublishDirectory, "_framework", "System.Private.CoreLib.dll.br");
+            Assert.FileDoesNotExist(result, secondAppPublishDirectory, "_framework", "System.Text.Json.dll.br");
+        }
+
         private static void AddWasmProjectContent(ProjectDirectory project, string content)
         {
             var path = Path.Combine(project.SolutionPath, "blazorwasm", "blazorwasm.csproj");
@@ -1021,7 +1073,6 @@ $@"<PropertyGroup>
                 return webFormattedHash.Substring(7);
             }
         }
-
 
         private void VerifyTypeGranularTrimming(MSBuildResult result, string blazorPublishDirectory)
         {

--- a/src/Components/WebAssembly/Sdk/src/targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets
+++ b/src/Components/WebAssembly/Sdk/src/targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets
@@ -509,7 +509,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ItemGroup>
       <_BlazorPublishBootResource
         Include="@(ResolvedFileToPublish)"
-        Condition="$([System.String]::Copy('%(RelativePath)').Replace('\','/').StartsWith('wwwroot/_framework')) AND '%(Extension)' != '.a'" />
+        Condition="$([System.String]::Copy('%(RelativePath)').Replace('\','/').StartsWith($(_BlazorFrameworkPublishPath.Replace('\', '/')))) AND '%(Extension)' != '.a'" />
     </ItemGroup>
 
     <GetFileHash Files="@(_BlazorPublishBootResource)" Algorithm="SHA256" HashEncoding="base64">

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/App.razor
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/App.razor
@@ -1,0 +1,1 @@
+<h1>@typeof(BlazorMultipleApps.FirstClient.Program)</h1>

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/BlazorMultipleApps.FirstClient.csproj
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/BlazorMultipleApps.FirstClient.csproj
@@ -1,0 +1,24 @@
+<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <StaticWebAssetBasePath>FirstApp</StaticWebAssetBasePath>
+    <RazorSdkDirectoryRoot>$(RazorSdkArtifactsDirectory)$(Configuration)\sdk-output\</RazorSdkDirectoryRoot>
+    <BlazorWebAssemblySdkDirectoryRoot>$(BlazorWebAssemblySdkArtifactsDirectory)$(Configuration)\sdk-output\</BlazorWebAssemblySdkDirectoryRoot>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\BlazorMultipleApps.Shared\BlazorMultipleApps.Shared.csproj" />
+  </ItemGroup>
+
+  <PropertyGroup Condition="'$(BinariesRoot)'==''">
+    <!-- In test scenarios $(BinariesRoot) is defined in a generated Directory.Build.props file -->
+    <BinariesRoot>$(RepoRoot)artifacts\bin\Microsoft.AspNetCore.Razor.Test.MvcShim.ClassLib\$(Configuration)\netstandard2.0\</BinariesRoot>
+  </PropertyGroup>
+
+  <!-- DO NOT add addition references here. This is meant to simulate a non-MVC library -->
+  <ItemGroup Condition="'$(BinariesRoot)'!=''">
+    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.Test.ComponentShim.dll"/>
+  </ItemGroup>
+
+</Project>

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/Program.cs
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/Program.cs
@@ -1,0 +1,15 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Text;
+
+namespace BlazorMultipleApps.FirstClient
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            GC.KeepAlive(typeof(System.Text.Json.JsonSerializer));
+        }
+    }
+}

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/wwwroot/css/app.css
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/wwwroot/css/app.css
@@ -1,0 +1,1 @@
+﻿/* First app.css */

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/wwwroot/index.html
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.FirstClient/wwwroot/index.html
@@ -1,0 +1,14 @@
+﻿<!DOCTYPE html>
+<html>
+
+<head>
+    <title>BlazorMultipleApps</title>
+    <base href="/FirstApp/" />
+    <link href="css/app.css" rel="stylesheet" />
+</head>
+
+<body>
+    <script src="_framework/blazor.webassembly.js"></script>
+</body>
+
+</html>

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/App.razor
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/App.razor
@@ -1,0 +1,1 @@
+<h1>@typeof(BlazorMultipleApps.SecondClient.Program)</h1>

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/BlazorMultipleApps.SecondClient.csproj
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/BlazorMultipleApps.SecondClient.csproj
@@ -1,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.BlazorWebAssembly">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+    <StaticWebAssetBasePath>SecondApp</StaticWebAssetBasePath>
+    <RazorSdkDirectoryRoot>$(RazorSdkArtifactsDirectory)$(Configuration)\sdk-output\</RazorSdkDirectoryRoot>
+    <BlazorWebAssemblySdkDirectoryRoot>$(BlazorWebAssemblySdkArtifactsDirectory)$(Configuration)\sdk-output\</BlazorWebAssemblySdkDirectoryRoot>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\BlazorMultipleApps.Shared\BlazorMultipleApps.Shared.csproj" />
+  </ItemGroup>
+
+  <PropertyGroup Condition="'$(BinariesRoot)'==''">
+    <!-- In test scenarios $(BinariesRoot) is defined in a generated Directory.Build.props file -->
+    <BinariesRoot>$(RepoRoot)artifacts\bin\Microsoft.AspNetCore.Razor.Test.MvcShim.ClassLib\$(Configuration)\netstandard2.0\</BinariesRoot>
+  </PropertyGroup>
+
+  <!-- DO NOT add addition references here. This is meant to simulate a non-MVC library -->
+  <ItemGroup Condition="'$(BinariesRoot)'!=''">
+    <Reference Include="$(BinariesRoot)\Microsoft.AspNetCore.Razor.Test.ComponentShim.dll"/>
+  </ItemGroup>
+
+</Project>

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/Program.cs
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/Program.cs
@@ -1,0 +1,12 @@
+using System;
+using System.Collections.Generic;
+
+namespace BlazorMultipleApps.SecondClient
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+        }
+    }
+}

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/wwwroot/css/app.css
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/wwwroot/css/app.css
@@ -1,0 +1,1 @@
+﻿/* Second app.css */

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/wwwroot/index.html
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.SecondClient/wwwroot/index.html
@@ -1,0 +1,14 @@
+﻿<!DOCTYPE html>
+<html>
+
+<head>
+    <title>SecondBlazorApp</title>
+    <base href="/SecondApp/" />
+    <link href="css/app.css" rel="stylesheet" />
+</head>
+
+<body>
+    <script src="_framework/blazor.webassembly.js"></script>
+</body>
+
+</html>

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.Server/BlazorMultipleApps.Server.csproj
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.Server/BlazorMultipleApps.Server.csproj
@@ -1,0 +1,14 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
+
+  <PropertyGroup>
+    <TargetFramework>net5.0</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\BlazorMultipleApps.SecondClient\BlazorMultipleApps.SecondClient.csproj" />
+    <ProjectReference Include="..\BlazorMultipleApps.FirstClient\BlazorMultipleApps.FirstClient.csproj" />
+    <ProjectReference Include="..\BlazorMultipleApps.Shared\BlazorMultipleApps.Shared.csproj" />
+  </ItemGroup>
+
+
+</Project>

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.Server/Program.cs
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.Server/Program.cs
@@ -1,0 +1,10 @@
+﻿namespace BlazorMultipleApps.Server
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            
+        }
+    }
+}

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.Shared/BlazorMultipleApps.Shared.csproj
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.Shared/BlazorMultipleApps.Shared.csproj
@@ -1,0 +1,7 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.1</TargetFramework>
+  </PropertyGroup>
+
+</Project>

--- a/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.Shared/WeatherForecast.cs
+++ b/src/Components/WebAssembly/Sdk/testassets/BlazorMultipleApps.Shared/WeatherForecast.cs
@@ -1,0 +1,17 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace BlazorMultipleApps.Shared
+{
+    public class WeatherForecast
+    {
+        public DateTime Date { get; set; }
+
+        public int TemperatureC { get; set; }
+
+        public string Summary { get; set; }
+
+        public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);
+    }
+}

--- a/src/Components/WebAssembly/Sdk/testassets/RestoreBlazorWasmTestProjects/RestoreBlazorWasmTestProjects.csproj
+++ b/src/Components/WebAssembly/Sdk/testassets/RestoreBlazorWasmTestProjects/RestoreBlazorWasmTestProjects.csproj
@@ -9,6 +9,7 @@
     <ProjectReference Include="..\blazorwasm\blazorwasm.csproj" />
     <ProjectReference Include="..\blazorwasm-minimal\blazorwasm-minimal.csproj" />
     <ProjectReference Include="..\blazorwasm-fxref\blazorwasm-fxref.csproj" />
+    <ProjectReference Include="..\BlazorMultipleApps.Server\BlazorMultipleApps.Server.csproj" />
   </ItemGroup>
 
 </Project>

